computervision/src/main/java/ComputerVisionInAction.kt

```diff
@@ -14,37 +15,36 @@ internal class ComputerVisionInAction(private val cvPortal: SensorPortal) : Comp
         cv = cvPortal.connect()
     }
 
-    private fun cvTokenize(rawData: List<Any>): Vision? {
-        val raw : String = rawData[0].toString()
-        return when {
+    private fun cvTokenize(rawData: String): Vision? {
+        val raw = JSONObject(rawData)
+        return when (raw.getInt("type") == 1) {
 
-            raw.startsWith('1') -> Vision.OCR(raw)
-            raw.startsWith('2') -> Vision.Scenery(raw)
-            raw.startsWith('3') -> Vision.Facial(raw)
-            raw.startsWith('4') -> Vision.Emotion(raw)
-            raw.startsWith('5') -> Vision.ObjectDetection(rawData)
+            raw.has("value") -> Vision.OCR(raw.optString("value"))
+            raw.has("value") -> Vision.Scenery(raw.optString("value"))
+            raw.has("value") -> Vision.Facial(raw.optString("value"))
+            raw.has("value") -> Vision.Emotion(raw.optString("value"))
+            raw.has("value") -> Vision.ObjectDetection(raw.optJSONArray("value"))
             else -> null // (corrupt reading, discard.) the russians did it for cv !
```

Comment:
this should be:
``` kotlin
return when (raw.getInt("type")) {
   1 -> Vision.OCR(raw.getString("value"))
   2 -> Vision.Scenery(raw.getString("value"))
   // and so on...
}
```
also, use `getString`, not `optString`; We are sure that there will be a value, otherwise it must be an error. [check this out](https://stackoverflow.com/questions/13790726/the-difference-between-getstring-and-optstring-in-json/13790789).

Also, you might define some constants (not necessarily in the same file) to avoid confusions, like:
```kotlin
const val OCR = 1
```
and use it like:
```kotlin
return when (raw.getInt("type")) {
   OCR -> Vision.OCR(raw.getString("value"))
   // ...
}
```

Reply:
I thought it would be more safe to use optString not to throw any exceptions if there's an otherwise probability? And for that, we will get a null instead(which is a sign for a possible occurred error),  so the process will still be running.

Okay!, changes -> done!

Reply:
You're actually right. However, if the message is corrupt, you wouldn't be able to obtain a `JsonObject` in the first place, maybe test handle this case?

Reply:
I see, okay, a try catch block is enough?

Reply:
I think so. And in the catch return null.

Reply:
All done till now. Thank you!

Reply:
Nice! You are welcome!